buildutils/src/upgrade-lab-dependencies.ts

```diff
@@ -0,0 +1,101 @@
+import fs from 'fs';
+import path from 'path';
+
+const PACKAGE_JSON_PATHS: string[] = [
+  'app/package.json',
+  'buildutils/package.json',
+  'package.json',
+  'packages/application-extension/package.json',
+  'packages/application/package.json',
+  'packages/console-extension/package.json',
+  'packages/docmanager-extension/package.json',
+  'packages/documentsearch-extension/package.json',
+  'packages/help-extension/package.json',
+  'packages/lab-extension/package.json',
+  'packages/notebook-extension/package.json',
+  'packages/terminal-extension/package.json',
+  'packages/tree-extension/package.json',
+  'packages/tree/package.json',
+  'packages/ui-components/package.json',
+];
+
+const DEPENDENCY_GROUP = '@jupyterlab';
+
+async function updatePackageJson(newVersion: string): Promise<void> {
+  const url = `https://raw.githubusercontent.com/jupyterlab/jupyterlab/v${newVersion}/jupyterlab/staging/package.json`;
+  const response = await fetch(url);
+
+  if (!response.ok) {
+    const errorMessage = `Failed to fetch package.json from ${url}. HTTP status code: ${response.status}`;
+    throw new Error(errorMessage);
+  }
+
+  const newPackageJson = await response.json();
+
+  for (const packageJsonPath of PACKAGE_JSON_PATHS) {
+    const filePath: string = path.resolve(packageJsonPath);
+    const existingPackageJson = JSON.parse(fs.readFileSync(filePath, 'utf-8'));
+
+    const newDependencies = {
+      ...newPackageJson.devDependencies,
+      ...newPackageJson.resolutions,
+    };
+
+    updateDependencyVersion(existingPackageJson, newDependencies);
+
+    fs.writeFileSync(filePath, JSON.stringify(existingPackageJson, null, 2));
```

Comment:
I tried that just now and noticed that it is missing a final new line so maybe:

```suggestion
    fs.writeFileSync(filePath, JSON.stringify(existingPackageJson, null, 2) + '\n');
```

?

Reply:
I think adding a newline isn’t necessary since our current package.json doesn’t have one, and even though if we have one our lint-check would flag it. Correct me If I am wrong.

Reply:
So we need a single new line and lack of it is actually flagged by precommit linter. We do not need/want a trailing new line, just the last line to end with new line symbol.